### PR TITLE
PI-1180 PI-1177 PI-1191 PI-1181 update UpgradePolicy

### PR DESCRIPTION
**Pull request by seansellek**

### CODE REVIEW

**Comment by davidyscott** [comment]
So... this is interesting with versioning...   WE've now created a situation where the client lib before this commit will break if used with a service more recent than the corresponding commit, and the client lib after this commit will break if used with an older service.  Seems pretty suboptimal?  But I don't have any useful suggestion to make so meh

**Comment by seansellek** [comment]
@davidyscott speaking of which, is there any versioning stuff I need to update? Haven't worked in this lib before.

**Comment by davidyscott** [comment]
@seansellek that's entirely @natke 's domain...

**Comment by natke** [comment]
@seansellek As this is a breaking change, you will need to update the major version

**Comment by seansellek** [comment]
@natke @4141done went ahead and made all the changes to UpgradePolicy here

**Comment by natke** [comment]
Can we also update the compatibility table at the bottom of the README with the ItemSense build #s?

**Comment by seansellek** [comment]
@natke tests added to ensure expected behaviour when property isn't set.

I will update with a build number when we have one.

File: pom.xml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <groupId>com.impinj</groupId>
     <artifactId>itemsense-client</artifactId>
     <packaging>jar</packaging>
-    <version>3.0.1-SNAPSHOT</version>
+    <version>4.0.0-SNAPSHOT</version>
```
Comment (natke) [inline pom.xml]: @seansellek Let's change the 2016r6 entry in the compatibility table to 4.0.0.

Comment (seansellek) [inline pom.xml]: Done!

**Pull request merged**